src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -2,23 +2,20 @@
 
 import org.skypro.skyshop.product.Product;
 
-import java.util.Iterator;
-import java.util.LinkedList;
+import java.util.*;
 
 public class ProductBasket {
-    LinkedList<Product> products = new LinkedList<>();
-    private int counter = 0;
+    Map<String, LinkedList<Product>> products = new HashMap<>();
```

Comment:
поля должны быть приватные всегда

Reply:
```suggestion
    Map<String, List<Product>> products = new HashMap<>();
```
также не забываем про полиморфизм